backend/src/services/sumo_access/table_access.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from typing import List
+
+import pyarrow as pa
+from fmu.sumo.explorer.objects import CaseCollection
+from sumo.wrapper import SumoClient
+
+from src.services.types.generic_types import TableMetaData
+
+from ._helpers import create_sumo_client_instance
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TableAccess:
+    def __init__(self, access_token: str, case_uuid: str, iteration_name: str):
+        self._sumo_client: SumoClient = create_sumo_client_instance(access_token)
+        self._case_uuid = case_uuid
+        self._iteration_name = iteration_name
+
+    def get_realization_tables_metadata(self, realization: int = 0) -> List[TableMetaData]:
+        """Get all table descriptions for a given realization"""
+
+        case_collection = CaseCollection(self._sumo_client).filter(uuid=self._case_uuid)
+        if len(case_collection) != 1:
+            raise ValueError(f"None or multiple sumo cases found {self._case_uuid=}")
+
+        case = case_collection[0]
+        table_collection = case.tables.filter(
+            realization=realization,
+            iteration=self._iteration_name,
+        )
+
+        return [
+            TableMetaData(
+                name=table.name,
+                tagname=table.tagname,
+                column_names=table.metadata.get("data", {}).get("spec", {}).get("columns", []),
+                file_format=table.format,
+            )
+            for table in table_collection
+        ]
+
+    def get_realization_table(
+        self,
+        table_metadata: TableMetaData,
+        realization: int = 0,
+    ) -> pa.Table:
+        """Get a pyarrow table for a given realization"""
+
+        case_collection = CaseCollection(self._sumo_client).filter(uuid=self._case_uuid)
+        if len(case_collection) != 1:
+            raise ValueError(f"None or multiple sumo cases found {self._case_uuid=}")
+
+        case = case_collection[0]
+
+        table_collection = case.tables.filter(
+            name=table_metadata.name,
+            tagname=table_metadata.tagname,
+            iteration=self._iteration_name,
+            realization=realization,
+        )
+        if len(table_collection) == 0:
```

Comment:
Interesting that `pylint` does not complain on this (https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/use-implicit-booleaness-not-len.html). Does it work with `if not table_collection`?

Reply:
yes. fixed